### ISSUE TITLE
Missing/Incomplete type for events prop

### ISSUE THREAD

**Issue by Odame** [SupremeTechnopriest/react-idle-timer#107]
**Version 4.2.0 brings typescript support and dynamic event binding to IdleTimer**, which is really great.

But currently, the events prop has its type specified as `string[]`.

While this works without any issue in typescript, I think it will be better if we has a union set of strings that match the supported events, or an `enum`. 

Here is an example that highlights the potential problem with specifying the type as `string[]`.

```jsx
import React, { Component, useCallback } from 'react'
import IdleTimer from 'react-idle-timer'
 
export default function App(){
   const onIdle = useCallback(() => console.log('user is idle'));
   const onActive = useCallback(() => console.log('user is active'));
   
   const events = [
        'wheel',
        'DOMouseScroll', // Notice the mispelling in DOMMouseScroll
        'mouseWheel',
        'touchmove',
        'MSPointerMove',
   ]
   return (
      <div>
        <IdleTimer
          element={document}
          onActive={onActive}
          onIdle={onIdle} 
          events={events}
       />
        {/* App logic here */}
      </div>
    )
}
```

The error due to mispelling in `DOMMouseScroll` will caught by typescript if this were to be an enum, for instance. Thus reducing frustration; I took me some time to figure out the issue with my code.

If this passes as valid issue, I can make a PR to resolve this, possibly this weekend. It will be a good first PR for me!

**Comment by SupremeTechnopriest** [comment]
Hey @Odame!

Feel free to add an ENUM type for events.  Thats probably a good idea.  The library is not written in Typescript.  Personally I find TS too quirky and slow.  So in order to support the TS community I manage the definition file manually.  

https://github.com/SupremeTechnopriest/react-idle-timer/blob/master/src/index.d.ts#L52

Feel free to open up a pull request and get your contributer badge!

**Comment by SupremeTechnopriest** [comment]
Hey @Odame,

I've gone ahead and added this to the next release.  It's not ready yet, but I wanted to let you know before you began any work on this.  So far I have the following events:

```ts
  enum EVENTS {
    'abort',
    'afterprint',
    'animationend',
    'animationiteration',
    'animationstart',
    'beforeprint',
    'beforeunload',
    'blur',
    'canplay',
    'canplaythrough',
    'change',
    'click',
    'contextmenu',
    'copy',
    'cut',
    'dblclick',
    'drag',
    'dragend',
    'dragenter',
    'dragleave',
    'dragover',
    'dragstart',
    'drop',
    'durationchange',
    'ended',
    'error',
    'focus',
    'focusin',
    'focusout',
    'fullscreenchange',
    'fullscreenerror',
    'hashchange',
    'input',
    'invalid',
    'keydown',
    'keypress',
    'keyup',
    'load',
    'loadeddata',
    'loadedmetadata',
    'loadstart',
    'message',
    'mousedown',
    'mouseenter',
    'mouseleave',
    'mousemove',
    'mouseover',
    'mouseout',
    'mouseup',
    'mousewheel',
    'offline',
    'online',
    'open',
    'pagehide',
    'pageshow',
    'paste',
    'pause',
    'play',
    'playing',
    'popstate',
    'progress',
    'ratechange',
    'resize',
    'reset',
    'scroll',
    'search',
    'seeked',
    'seeking',
    'select',
    'show',
    'stalled',
    'storage',
    'submit',
    'suspend',
    'timeupdate',
    'toggle',
    'touchcancel',
    'touchend',
    'touchmove',
    'touchstart',
    'transitionend',
    'unload',
    'volumechange',
    'waiting',
    'wheel'
  }
```

Let me know if you think I have missed anything and I will add it before release (hopefully sometime next week)

**Comment by SupremeTechnopriest** [comment]
This list can probably be reduced down quite a bit since a lot of them will have triggered a `mousemove` or `mousedown` event before firing.

**Comment by Odame** [comment]
@SupremeTechnopriest 
I went through the events enum in the https://github.com/SupremeTechnopriest/react-idle-timer/issues/107#issuecomment-623048656. And I think it's quite exhaustive.

**Comment by SupremeTechnopriest** [comment]
@Odame Cool Cool!  I will release this with the functional component hook patch this week!

**Comment by SupremeTechnopriest** [comment]
@Odame Hey hey! This is now published to npm as version `4.3.0` alongside the `useIdleTimer` hook.  Let me know if I messed anything up!
